tools/shell.nix

```diff
@@ -0,0 +1,70 @@
+# Development shell environment for Randovania on Nix.
+#
+# This will:
+# - create the venv and install all requirements, if it doesn't already exist
+# - set up $LD_LIBRARY_PATH with the system libraries needed for RDV
+# - define $RANDOVANIA_PATH as the root of the RDV source checkout
+# - define `randovania` as a shell function that runs RDV from the source root
+#
+# So, basic usage is:
+# $ nix-shell tools/shell.nix
+# [...long wait while it installs requirements, if this is the first time]
+# [...do some coding...]
+# $ rdv
+#
+# Or, if you just want to run rdv as a one-off:
+# $ nix-shell tools/shell.nix --run rdv
+
+{ pkgs ? import <nixpkgs> {} }:
+
+let
+  libs = with pkgs; [
+    dbus
+    fontconfig
+    freetype
+    glib
+    libgcc.lib
+    libGL
+    libxkbcommon
+    libz
+    wayland
+    xcb-util-cursor
+    xorg.libxcb
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+    xorg.libX11
```

Comment:
curios, why are both wayland and x11 deps specified? Just as a fallback if someone runs this on an x11 system?

Reply:
Yeah. RDV uses both (since it supports both wayland and X11), so both need to be in the closure.